lib/make-disk-image.nix

```diff
@@ -0,0 +1,56 @@
+{ nixosConfig
+, diskoLib
+, pkgs ? nixosConfig.pkgs
+, lib ? pkgs.lib
+, name ? "${nixosConfig.config.networking.hostName}-disko-images"
+}:
+let
+  cleanedConfig = diskoLib.testLib.prepareDiskoConfig nixosConfig.config diskoLib.testLib.devices;
+  systemToInstall = nixosConfig.extendModules {
+    modules = [{
+      disko.devices = lib.mkForce cleanedConfig.disko.devices;
+      boot.loader.grub.devices = lib.mkForce cleanedConfig.boot.loader.grub.devices;
+    }];
+  };
+in
+pkgs.vmTools.runInLinuxVM (pkgs.runCommand name {
+  buildInputs = with pkgs; [
+    systemdMinimal
+    nixos-install-tools
+    nix
+    utillinux
+  ];
```

Comment:
When people have common hooks, we probably should expose an extraPackages parameter?

Reply:
Maybe not needed.